sdks/python/apache_beam/io/gcp/bigquery_file_loads.py

```diff
@@ -90,9 +89,12 @@ def _generate_file_prefix(pipeline_gcs_location):
 
 
 def _make_new_file_writer(file_prefix, destination):
-  if isinstance(destination, bigquery_api.TableReference):
-    destination = '%s:%s.%s' % (
-        destination.projectId, destination.datasetId, destination.tableId)
+  destination = bigquery_tools.get_hashable_destination(destination)
+
+  # Windows does not allow : on filenames. Replacing with underscore.
+  # Other disallowed characters are:
+  # https://docs.microsoft.com/en-us/windows/desktop/fileio/naming-a-file
+  destination = destination.replace(':', '_')
```

Comment:
Suggestion: replace the `:` with a dot, '.', since it's consistent with BigQuery standard SQL (see previous comment).